chia/timelord/timelord_api.py

```diff
@@ -75,6 +75,19 @@ async def new_peak_timelord(self, new_peak: timelord_protocol.NewPeakTimelord) -
             if self.timelord.last_state.peak.reward_chain_block.get_hash() == new_peak.reward_chain_block.get_hash():
                 log.info("Skipping peak, already have.")
             else:
+                if (
+                    self.timelord.last_state.peak.reward_chain_block.total_iters
+                    >= new_peak.reward_chain_block.total_iters
+                ):
```

Comment:
shouldn't you check for equal weight as well? All we know at this point is that the hash is different (checked in line 75) and that the weight is not less (checked on line 56). But, as I read it, it's still possible for `new_peak` to have a *lower* weight here.

Reply:
yes thanks, fixed